src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1163,6 +1153,32 @@ private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLen
             return this;
         }
 
+        private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLength, T[] values)
+        {
+            if (values == null)
+                throw new ArgumentNullException(nameof(values));
```

Comment:
Much of corelib would do `ThrowHelper.ThrowArgumentNullException` to reduce IL. but I see we don't do it anywhere in this class so I guess this is fine.